website/utils/url_utils.ts

```diff
@@ -43,6 +87,67 @@ const tryDecodeURLList = (url: string, delim = ","): string[] | undefined => {
   return urls;
 };
 
+/**
+ * Parse a string list of comma-separated key:value pairs into
+ * a key-value object.
+ *
+ * @param data The string to parse. Expected to be in the format
+ * "key1:value1,key2:value2,...". Colons and commas in keys or values
+ * should be encoded using `encodeURIComponent`.
+ * @returns An object with the parsed key-value pairs. Key and value strings
+ *  will be decoded using `decodeURIComponent`.
+ */
+export function parseKeyValueList(data: string): Record<string, string> {
+  if (data === "") {
+    return {};
+  }
+  const result: Record<string, string> = {};
+  const keyValuePairs = data.split(",");
+  for (const pair of keyValuePairs) {
+    const splitIndex = pair.indexOf(":");
+    const key = pair.slice(0, splitIndex);
+    const value = pair.slice(splitIndex + 1);
+    result[decodeURIComponent(key).trim()] = decodeURIComponent(value).trim();
+  }
+  return result;
+}
+
+function parseFloat(value: string | undefined): number | undefined {
+  if (value === undefined) {
+    return undefined;
+  }
+  const number = Number.parseFloat(value);
+  return Number.isNaN(number) ? undefined : number;
+}
+
+export function deserializeViewerChannelSetting(
+  channelIndex: number,
+  jsonState: ViewerChannelSettingJson
+): ViewerChannelSetting {
+  // Missing/undefined fields should be handled downstream.
+  const result: ViewerChannelSetting = {
+    match: channelIndex,
+    enabled: jsonState.ven === "1",
+    surfaceEnabled: jsonState.sen === "1",
+    isovalue: parseFloat(jsonState.isv),
+    surfaceOpacity: parseFloat(jsonState.isa),
+    colorizeEnabled: jsonState.clz === "1",
+    colorizeAlpha: parseFloat(jsonState.cza),
+  };
+  if (jsonState.col) {
+    if (HEX_COLOR_REGEX.test(jsonState.col)) {
+      result.color = jsonState.col;
+    }
+  }
+  if (jsonState.lut) {
+    if (LUT_REGEX.test(jsonState.lut)) {
+      const [min, max] = jsonState.lut.split(":");
+      result.lut = [min.trim(), max.trim()];
+    }
+  }
```

Comment:
More because I think it's cool than because it produces better or more readable code: TypeScript knows that `&&` is short-circuiting, and will narrow types on the right-hand side based on the left:
```typescript
if (jsonState.col && HEX_COLOR_REGEX.test(jsonState.col)) {
  // ...
}
```